### PR TITLE
[Docs] Document how a waiter works

### PR DESCRIPTION
**Pull request by knownasilya**
A simple explanation of how a waiter works would help anyone looking to write one.

### CODE REVIEW

File: packages/ember-testing/lib/test/waiters.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ const callbacks = [];
 /**
    This allows ember-testing to play nicely with other asynchronous
    events, such as an application that is waiting for a CSS3
-   transition or an IndexDB transaction.
+   transition or an IndexDB transaction. The waiter runs periodically
```
Comment (rwjblue) [inline packages/ember-testing/lib/test/waiters.js]: I would slightly clarify that it is ran periodically after async helpers (i.e. `click`, `andThen`, `visit`, etc) have executed prior to executing the next async step.

Comment (knownasilya) [inline packages/ember-testing/lib/test/waiters.js]: Got it.

Do all waiters just run or do you use `app.testHelpers.wait()` in your custom asyncHelper to do a wait check?

Comment (rwjblue) [inline packages/ember-testing/lib/test/waiters.js]: The work is done by way of `wait()`, but all async helpers run `app.testHelpers.wait()` at the end by default (you shouldn't have to `return app.testHelpers.wait()` manually in an async test helper).

File: packages/ember-testing/lib/test/waiters.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ const callbacks = [];
    This allows ember-testing to play nicely with other asynchronous
    events, such as an application that is waiting for a CSS3
    transition or an IndexDB transaction. The waiter runs periodically
-   until the returning result is truthy.
+   after all async helpers (i.e. `click`, `andThen`, `visit`, etc) have executed,
```
Comment (rwjblue) [inline packages/ember-testing/lib/test/waiters.js]: (sorry) `after all` -> `after each` 😺

File: packages/ember-testing/lib/test/waiters.js (hunk 1 above)
Comment (knownasilya) [inline packages/ember-testing/lib/test/waiters.js]: Updated

**Comment by locks** [comment]
@rwjblue is this good to go?

**Comment by rwjblue** [comment]
Yes, this is good, just needs a squashing now...

**Comment by locks** [comment]
@rwjblue is there a problem with using GitHub interface's squashing mechanism? It makes something weird to the git history, right?

**Comment by rwjblue** [comment]
It breaks my CHANGELOG generation script so I disabled it until we can get it fixed.

**Pull request closed without merging**